### PR TITLE
added a review page with an approval list

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

File: backend/src/models/v2/Approval.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+)
+
+ApprovalRequestSchema.plugin(MongooseDelete, {
+  overrideMethods: 'all',
+  deletedBy: true,
+  deletedByType: Schema.Types.ObjectId,
+})
+
+const ApprovalRequestModel = model<ApprovalRequestInterface>('v2_Release', ApprovalRequestSchema)
```
Comment (JR40159) [inline backend/src/models/v2/Approval.ts:42]: V2_Approval?

File: frontend/types/types.ts [hunk 2]
```diff
@@ -410,3 +410,13 @@ export const SchemaKind = {
 } as const
 
 export type SchemaKindKeys = (typeof SchemaKind)[keyof typeof SchemaKind]
+
+export interface ApprovalRequestInterface {
+  model: string
+  release: string
+  kind: string
+  isActive: boolean
+  entities: Array<string>
```
Comment (a3957273) [inline frontend/types/types.ts:419]: No entities here anymore.

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: backend/src/routes/v2/approval/getApprovals.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+export const getApprovals = [
+  bodyParser.json(),
+  async (req: Request, res: Response<GetApprovalsResponse>) => {
+    const _ = parse(req, getApprovalsSchema)
+    const { isActive } = req.query
+    if (isActive !== 'true' && isActive !== 'false') {
+      throw BadReq('isActive should be true or false')
+    }
+    const active = req.query.isActive === 'true'
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply backend/src/routes/v2/approval/getApprovals.ts:31]: This is being handled in another ticket

File: backend/src/routes/v2/approval/getApprovals.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    approvals: Array<ApprovalRequestInterface>
+  }
+}
+
+export const getApprovals = [
+  bodyParser.json(),
+  async (req: Request, res: Response<GetApprovalsResponse>) => {
+    const _ = parse(req, getApprovalsSchema)
+    const { isActive } = req.query
+    if (isActive !== 'true' && isActive !== 'false') {
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply backend/src/routes/v2/approval/getApprovals.ts:28]: This is being handled in another ticket

File: backend/src/routes/v2/approval/getApprovals.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { Request, Response } from 'express'
+import { z } from 'zod'
+
+import { parse } from '../../../middleware/validate.js'
+import { ApprovalRequestInterface } from '../../../models/v2/Approval.js'
+import { BadReq } from '../../../utils/v2/error.js'
+
+export const getApprovalsSchema = z.object({
+  query: z.object({
+    isActive: z.string({
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply backend/src/routes/v2/approval/getApprovals.ts:11]: This is being handled in another ticket

**Review by a3957273 — APPROVED** [review]

**Pull request merged**